oemof/solph/linear_constraints.py

```diff
@@ -538,15 +539,17 @@ def add_storage_balance(model, block):
     # set cap of last timesteps to fixed value of cap_initial
     t_last = len(model.timesteps)-1
     for e in block.uids:
-      block.cap[e, t_last] = cap_initial[e]
-      block.cap[e, t_last].fix()
+        if cap_initial[e] is not None:
+            block.cap[e, t_last] = cap_initial[e]
+            block.cap[e, t_last].fix()
 
     def storage_balance_rule(block, e, t):
-        # TODO:
-        #   - include time increment
+        # TODO: include time increment
         expr = 0
         if(t == 0):
-            expr += block.cap[e, t] - cap_initial[e]
+            t_last = len(model.timesteps)-1
+            expr += block.cap[e, t]
+            expr += - block.cap[e, t_last] * (1 - cap_loss[e])
```

Comment:
The only difference can be found on line 552 of the new (green) version. Because I added the self-discharge to the first timestep. Now it will work for years without violating the balance. But than we have to discuss, if we replace `block.cap[e, t_last]` in the lines 543 and 544 of the new version with `block.cap[e, 0]` as `cap_initial[e]` sounds like the state of charge at timestep 0. But first I wanted to change as few as possible.